src/lazy/any_encoding.rs

```diff
@@ -89,6 +89,18 @@ pub enum LazyRawAnyVersionMarkerKind<'top> {
     Binary_1_1(LazyRawBinaryVersionMarker_1_1<'top>),
 }
 
+impl<'top> LazyRawAnyVersionMarker<'top> {
+    pub fn encoding(&self) -> IonEncoding {
+        use crate::lazy::any_encoding::LazyRawAnyVersionMarkerKind::*;
+        match self.encoding {
+            Text_1_0(_) => TextEncoding_1_0.encoding(),
+            Binary_1_0(_) => BinaryEncoding_1_0.encoding(),
+            Text_1_1(_) => TextEncoding_1_1.encoding(),
+            Binary_1_1(_) => BinaryEncoding_1_1.encoding(),
+        }
+    }
+}
```

Comment:
🗺️ I plumbed the `IonEncoding` from the reader up through all of the raw stream item types (version marker, value, e-expression, end of stream), making it easy to check the encoding of any item during iteration.